File: download_model.js
import { pipeline, env } from '@xenova/transformers';
import * as fs from 'fs';
import * as path from 'path';

async function downloadModel() {
  env.useBrowserCache = false; // Disable browser cache for local downloading
  env.backends.default = 'local'; // Use local backend for downloading

  const modelId = 'Xenova/distilbert-base-uncased-finetuned-consumer-complaint-sentiment';
  const modelDir = path.join(process.cwd(), 'models');

  // Create the models directory if it doesn't exist
  if (!fs.existsSync(modelDir)) {
    fs.mkdirSync(modelDir, { recursive: true });
  }

  console.log(`Downloading model ${modelId} to ${modelDir}...`);

  try {
    // Create the pipeline instance
    const pipe = await pipeline(
      'text-classification',
      modelId,
      {
        quantized: true,
        progress_callback: (data) => {
            console.log(data)
        }
      },
    );

    console.log('Model downloaded successfully.');
  } catch (error) {
    console.error('Error downloading model:', error);
  }
}

downloadModel();

File: download_model.py
from huggingface_hub import hf_hub_download
import os

model_id = "Xenova/distilbert-base-uncased-finetuned-consumer-complaint-sentiment"
model_dir = "models"

if not os.path.exists(model_dir):
    os.makedirs(model_dir)

filenames = [
    "config.json",
    "model.safetensors",
    "tokenizer.json",
    "tokenizer_config.json",
    "vocab.txt",
    "special_tokens_map.json",
]

for filename in filenames:
    try:
        hf_hub_download(
            repo_id=model_id,
            filename=filename,
            local_dir=model_dir,
            local_dir_use_symlinks=False,
        )
        print(f"Downloaded {filename} successfully.")
    except Exception as e:
        print(f"Error downloading {filename}: {e}")